### PR TITLE
Update Secure XGBoost example w.r.t. XGBoost's code changes

### PR DESCRIPTION
**Pull request by ZiyueXu77**
Fixes # .

### Description

Update NVFlare's secure XGB example according to latest XGB code changes

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

**Comment by YuanTingHsieh** [comment]
Please also take any changes in https://github.com/NVIDIA/NVFlare/pull/2673 that is necessary, thanks

File: examples/advanced/xgboost_secure/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+nvflare~=2.4.2rc
```
Comment (YuanTingHsieh) [inline examples/advanced/xgboost_secure/requirements.txt:1]: 2.5.0rc?

File: examples/advanced/xgboost_secure/utils/prepare_data_horizontal.py [hunk 2]
```diff
@@ -70,8 +70,10 @@ def main():
     df_valid = df.iloc[int(0.8 * rows_total) :, :]
 
     for site in range(args.site_num):
-        # sort df_train by feature 2
-        # df_train = df_train.sort_values(by=2)
+        # sort df_train by feature 7,
+        # creating distribution shift between sites
+        # to illustrate the horizontal split quantile difference
+        df_train = df_train.sort_values(by=7)
```
Comment (YuanTingHsieh) [inline examples/advanced/xgboost_secure/utils/prepare_data_horizontal.py:76]: why feature 7?

File: examples/advanced/xgboost_secure/README.md [hunk 3]
```diff
@@ -43,16 +49,44 @@ For more details regarding federated XGBoost and the interface-plugin design, pl
...
+The running time of each job depends mainly on the encryption workload. 
+
+## Results
+Comparing the AUC results with centralized baseline, we have four observations:
 1. The performance of the model trained with homomorphic encryption is identical to its counterpart without encryption.
-2. Vertical federated learnings have identical performance as the centralized baseline.
-3. Horizontal federated learnings have performance slightly different from the centralized baseline. This is because under horizontal FL, the local histogram quantiles are based on the local data distribution, which may not be the same as the global distribution.
+2. Vertical federated learning (both secure and non-secure) have identical performance as the centralized baseline.
+3. Horizontal federated learning (both secure and non-secure) have performance slightly different from the centralized baseline. This is because under horizontal FL, the local histogram quantiles are based on the local data distribution, which may not be the same as the global distribution.
+4. GPU leads to different results compared to CPU, which is expected as the GPU involves some data conversions.
```
Comment (YuanTingHsieh) [inline examples/advanced/xgboost_secure/README.md:69]: Do we need to explain in the script we just use default "cuda:0" for all clients?
In the documentatio we might need to show them how they can configure this.

@SYangster

File: examples/advanced/xgboost_secure/train_standalone/train_federated.py [hunk 4]
```diff
@@ -23,9 +24,34 @@
...
+        help="Path to training data folder",
+    )
+    parser.add_argument(
+        "--data_test_file", type=str, default="/tmp/nvflare/xgb_dataset/test.csv", help="Path to testing data file"
+    )
+    parser.add_argument(
+        "--out_path",
+        type=str,
+        default="/tmp/nvflare/xgb_exp/base",
+        help="Output path for the data split file",
```
Comment (YuanTingHsieh) [inline examples/advanced/xgboost_secure/train_standalone/train_federated.py:52]: suggest  "/tmp/nvflare/xgboost_secure/train_standalone" , this is minor, we just need to make sure all the XGBoost related examples not using conflicting default path

**Review by YuanTingHsieh — COMMENTED** [review]
Mostly LGTM

File: examples/advanced/xgboost_secure/requirements.txt (hunk 1 above)
Comment (ZiyueXu77) [reply examples/advanced/xgboost_secure/requirements.txt:1]: yes, corrected

File: examples/advanced/xgboost_secure/utils/prepare_data_horizontal.py (hunk 2 above)
Comment (ZiyueXu77) [reply examples/advanced/xgboost_secure/utils/prepare_data_horizontal.py:76]: just an arbitrary choice

File: examples/advanced/xgboost_secure/README.md (hunk 3 above)
Comment (ZiyueXu77) [reply examples/advanced/xgboost_secure/README.md:69]: yes we can further update readme after adding the nvflare jobs

File: examples/advanced/xgboost_secure/train_standalone/train_federated.py (hunk 4 above)
Comment (ZiyueXu77) [reply examples/advanced/xgboost_secure/train_standalone/train_federated.py:52]: updated the defults (the actual out_dir remain unchanged in the run_training_standalone script)

**Review by YuanTingHsieh — APPROVED** [review]

**Pull request merged**